core/iwasm/libraries/debug-engine/debug_engine.c

```diff
@@ -960,8 +995,15 @@ wasm_debug_instance_kill(WASMDebugInstance *instance)
 
     while (exec_env) {
         wasm_cluster_thread_send_signal(exec_env, WAMR_SIG_TERM);
+        if (instance->current_state == APP_STOPPED) {
+            /* Resume all threads so they can receive the TERM signal */
+            exec_env->current_status->running_status = STATUS_RUNNING;
+            os_cond_signal(&exec_env->wait_cond);
```

Comment:
Should we wrap os_cond_signal with `os_mutex_lock(&exec_env->wait_lock)` and `os_mutex_unlock(&exec_env->wait_lock)`

Reply:
Done